### DISCUSSION TITLE
sudo nala upgrade && echo ---- && flatpak update && echo ---- && am -u

### DISCUSSION THREAD

**Discussion by Martin15135215** [ivan-hc/AM#74] [Q&A]
So I have this one-liner `sudo nala upgrade && echo ---- && flatpak update && echo ---- && am -u
` to upgrade all my packages on my Debian based system. Is there something to know about `am -u`? Does it also update itself, or do I need to `sudo nala upgrade && echo ---- && flatpak update && echo ---- && am -s && echo ---- && am -u
`?

**Comment by ivan-hc** [comment]
```
am -i topgrade
topgrade
```
- to update only AM, do  `am -s`;
-  to update AM and all programs managed by it, do `am -u`;
-  to update everything (AM, nala, flatpak...), `topgrade`.

More info at https://github.com/topgrade-rs/topgrade

**Reply by Martin15135215** [reply]
Thanks for the clarification! topgrade is an interesting project, I hope they find a [maintainer](https://github.com/topgrade-rs/topgrade#maintainers-wanted).
